src/vision/status_quo/barbara_battles_with_futures_unordered.md

```diff
@@ -0,0 +1,171 @@
+# 😱 Status quo stories: Barbara battles buffered items
+
+[How To Vision: Status Quo]: ../how_to_vision/status_quo.md
+[the raw source from this template]: https://raw.githubusercontent.com/rust-lang/wg-async-foundations/master/src/vision/status_quo/template.md
+[`status_quo`]: https://github.com/rust-lang/wg-async-foundations/tree/master/src/vision/status_quo
+[`SUMMARY.md`]: https://github.com/rust-lang/wg-async-foundations/blob/master/src/SUMMARY.md
+[open issues]: https://github.com/rust-lang/wg-async-foundations/issues?q=is%3Aopen+is%3Aissue+label%3Astatus-quo-story-ideas
+[open an issue of your own]: https://github.com/rust-lang/wg-async-foundations/issues/new?assignees=&labels=good+first+issue%2C+help+wanted%2C+status-quo-story-ideas&template=-status-quo--story-issue.md&title=
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+### Mysterious timeouts
+
+Barbara is working on her [YouBuy] server and is puzzling over a strange bug report. She is encountering users reporting that their browser connection is timing out when they connect to [YouBuy]. Based on the logs, she can see that they are timing out in the `do_select` function:
+
+```rust
+async fn do_select<T>(database: &Database, query: Query) -> Result<Vec<T>> {
+    let conn = database.get_conn().await?;
+    conn.select_query(query).await
+}
+```
+
+Some of the time, there is some kind of massive delay in between the `get_conn` method opening a connection and the call to `select_query`. But why? She has metrics that show that the CPU is largely idle, so it's not like the cores are all occupied.
+
+She looks at the caller of `do_select`, which is a function `do_work`:
+
+```rust
+async fn do_work(database: &Database) {
+    let work = do_select(database, FIND_WORK_QUERY)?;
+    stream::iter(
+        work
+            .into_iter()
+            .map(|item| do_select(database, work_from_item(item)).await)
+            .buffered(5)
+            .for_each(|work_item| process_work_item(database, work_item))
+    ).await;
```

Comment:
I think the nesting here is wrong. You probably want `stream::iter(work).map(...).buffered(5).for_each(...).await`

Reply:
Yeah, that seemed off to me too, which is partly why I put the indentation into it, to try and understand what was going on.

`Stream::iter` simply creates a stream from an iterator, right? 

And presumably one cannot in fact execute `await` inside a closure, so there is a missing `async move { }` block or something.

Reply:
Right, the `.await` needs to be removed as well. The `.buffered` combinator wants the stream item to be a future, so not awaiting it is the right thing to do here.

Reply:
And yes, `stream::iter` takes an `IntoIterator` and returns a `Stream`.